local-path-cleaner/blog_post.md

```diff
@@ -0,0 +1,619 @@
+---
+title: Addressing the Limitations of Local Path Provisioner in Kubernetes
+published: false
+description: In this post we will discuss how to manage ephemeral and semi-persistent local storage in Kubernetes by combining different Kubernetes features such as local ephemeral storage, filesystem quotas, and generic ephemeral volumes to address the limitations of local path provisioner.
+tags: kubernetes, cloud, devops, infrastructure
+cover_image: https://dev-to-uploads.s3.amazonaws.com/uploads/articles/3zcyrttro68enzhm57rt.png
+---
+
+## Temporary Storage in Kubernetes
+
+In Kubernetes, containers are ephemeral and stateless by default, allowing for easy scaling and management. Some workloads might require storage for temporary files, however. In vanilla Kubernetes, you are presented with the following options:
```

Comment:
I think there's more than

> require storage for temporary files

iiuc the question you're setting up is "How best to manage files over the different lifetimes of nodes, pods & disks?"

`emptyDir` follows the pod's lifecycle vs `hostPath` & `local` which follow the node's. There's a third (e.g. ebs) which have their own lifecycle.

... maybe `local` is actually a 4th, somewhere between the pod & the node, b/c it requires something to statically provision the PV. That thing could I guess also remove the PV before the node is terminated.